src/Symfony/Component/Security/Core/Exception/CustomUserMessageAuthenticationException.php

```diff
@@ -57,22 +57,24 @@ public function getMessageData()
 
     /**
      * {@inheritdoc}
+     *
+     * @param bool $isCalledFromOverridingMethod Must be set to true when called from an overriding method
+     *
+     * @return string|array Returns an array when $isCalledFromOverridingMethod is set to true
      */
     public function serialize()
     {
-        return serialize([
-            parent::serialize(),
-            $this->messageKey,
-            $this->messageData,
-        ]);
+        return serialize([$this->messageKey, $this->messageData, parent::serialize(true)]);
+
+        return $this->doSerialize($serialized, \func_num_args() ? \func_get_arg(0) : null);
     }
 
     /**
      * {@inheritdoc}
      */
     public function unserialize($str)
     {
-        list($parentData, $this->messageKey, $this->messageData) = unserialize($str);
+        list($this->messageKey, $this->messageData, $parentData) = \is_array($str) ? $str : unserialize($str);
```

Comment:
we should keep the original order, to preserve compat with existing serialized payloads